### PR TITLE
scripts: allow to display a script without focusing

### PR DESCRIPTION
**Pull request by aryangupta701**
- fixed focus issue for client-side recording now the script only grabs focus if it is a text-based script

### CODE REVIEW

**Comment by aryangupta701** [comment]
@thc202 Let me know if there is any better way to fix this issue.

**Comment by thc202** [comment]
This you push the correct changes? This is just removing it.

**Comment by aryangupta701** [comment]
The following code is already present there and until now this was useless. 
https://github.com/zaproxy/zap-extensions/blob/d0b147fbfc6db50c402bb0b96e962874db0087dc/addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java#L445-L448

**Comment by aryangupta701** [comment]
Wait, why do we want to call the setTabFocus method twice? Is there any reason for this?

**Comment by thc202** [comment]
It was not useless, see zaproxy/zap-extensions@abc6ad051e44f27352a2422a2d0830eddd6e3fa0. But yeah, we should not move it inside the if either and we shouldn't remove it either, we might need to make it optional instead.

**Comment by aryangupta701** [comment]
ohh got it. Thank you, will find a better solution now.

**Comment by aryangupta701** [comment]
This is the correct way of doing this? @thc202

**Comment by thc202** [comment]
As the first step it is but we also need to change the calling code to use it, maybe the fix will not be that trivial…

**Comment by aryangupta701** [comment]
I think we will also need to make an overloading function in this interface. Am I thinking in the right direction?  https://github.com/aryangupta701/zaproxy/blob/main/zap/src/main/java/org/zaproxy/zap/extension/script/ScriptUI.java#L132

**Comment by thc202** [comment]
Yes, that's the path but we should check first if there's a better one (like changing Zest somehow to avoid refreshing the script each time).

**Comment by aryangupta701** [comment]
may be we can trigger a display when the client recording turns off

**Comment by aryangupta701** [comment]
But I think that would be a bad option because user will not be able to verify if his interactions are being recorded or not

**Comment by aryangupta701** [comment]
> Yes, that's the path but we should check first if there's a better one (like changing Zest somehow to avoid refreshing the script each time).

Let's go with this option only. Because the rest other might spoil the user experience.

**Comment by aryangupta701** [comment]
okay sure

**Comment by aryangupta701** [comment]
@thc202 Should I start working on this? Other PRs might take some more time to merge because of the migrations going on and that's absolutely fine.

**Comment by thc202** [comment]
That sounds good to me.

**Comment by aryangupta701** [comment]
> Yes, that's the path but we should check first if there's a better one (like changing Zest somehow to avoid refreshing the script each time).

Should I go with this option?

**Comment by thc202** [comment]
I just checked and there's not really other way than changing the interface/core to allow to display the script without setting focus.

**Comment by aryangupta701** [comment]
okay will start it asap.

File: addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java [hunk 1]
```diff
@@ -448,7 +452,7 @@ public void setScript(ScriptWrapper script) {
             setTabFocus();
         }
 
-        if (!isTabVisible()) {
+        if (allowFocus && !isTabVisible()) {
```
Comment (thc202) [inline addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java:455]: Probably better to wrap both if statements that set focus, or just return earlier.

File: addOns/scripts/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 ### Changed
+- Focus only on Text-Based scripts to avoid recording issues with client scripts.
```
Comment (thc202) [inline addOns/scripts/CHANGELOG.md:8]: Should be tweaked to mention instead that it's possible to display a script without focusing.

Comment (thc202) [reply addOns/scripts/CHANGELOG.md:8]: PR title should also be updated.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**